### PR TITLE
Added a cli utility for removal of cached MLflow Conda environments.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
Added cli commands for listing and removing of Conda environments created by MLFlow.

When MLflow executes a project with associated conda environment file it will create a new conda
environment defined by this file. This environment is cached based on the content of the conda
    environment yaml file. This means it is only recreated if the project dependencies change.

    There are two reasons why you might need to rmove these files:

    1. To save disk space. The environments are never removed and can consume significant amount of
       disk space.
    2. To force recreation of the environment in cases when the caching mechanism failed. This
       happens e.g. when the conda file includes a dependency without specified version.

### CODE REVIEW

File: mlflow/utils/cli.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    When MLflow executes a project with associated conda environment file it will create a new conda
+    environment defined by this file. This environment is cached based on the content of the conda
+    environment yaml file. This means it is only recreated if the project dependencies change.
+
+    There are two reasons why you might need to remove these files:
+
+    1. To save disk space. The environments are never removed and can consume significant amount of
+       disk space.
+    2. To force recreation of the environment in cases when the caching mechanism failed. This
```
Comment (smurching) [inline mlflow/utils/cli.py]: Tried to dig into the caching problem a bit (created https://github.com/mlflow/mlflow/issues/525 to track it) - I think having this utility is useful regardless to free up disk space (as the docstring mentions).

**Review by smurching — APPROVED** [review]
Thanks @tomasatdatabricks this looks good, should we add docs for this CLI somewhere? Maybe to [this file](https://github.com/mlflow/mlflow/blob/781ce394953c7857b19c84dc937964a806b62798/docs/source/cli.rst)? Happy to have them be added in a follow-up.

File: mlflow/utils/cli.py (hunk 1 above)
Comment (tomasatdatabricks) [reply mlflow/utils/cli.py]: Thanks!

**Comment by tomasatdatabricks** [comment]
Hey @smurching, thanks for the review! Regarding the documentation, I was not sure to what level we want promote / document this feature. I think we can leave it for a followup PR.

**Comment by juntai-zheng** [comment]
Closed as part of old PR cleanup.

**Pull request closed without merging**